src/rpc/blockchain.cpp

```diff
@@ -1612,6 +1613,304 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
     return ret;
 }
 
+static CTransactionRef GetTransactionChecked(const uint256& hash)
+{
+    CTransactionRef tx_out;
+    uint256 hashBlock;
+    if (!GetTransaction(hash, tx_out, Params().GetConsensus(), hashBlock, false)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, std::string(fTxIndex ? "No such mempool or blockchain transaction"
```

Comment:
Surfacing this error seems like the the wrong error handling strategy. In the context this is used, the transaction should never be in the mempool and it should *always* be found if the txindex is present.

I think it would be better for the RPC to raise an RPC error that the txindex is required if it is not present, and an RPC error for an unexpected internal error (txindex is corrupt) otherwise.